Sources/NIO/EventLoop.swift

```diff
@@ -499,6 +502,16 @@ extension NIODeadline {
 }
 
 extension EventLoop {
+    /// Execute the given function and synchronously complete the given `EventLoopPromise` (if not `nil`).
+    @inlinable
+    public func executeAndComplete<T>(_ promise: EventLoopPromise<T>?, _ body: () throws -> T) {
+        do {
+            promise?.succeed(try body())
```

Comment:
Sadly you still can't use the `?` operator: you've introduced the same bug that was in #1746. Here again, if the promise is `nil`, the call to `try body()` will never be made. We need the code to look as it did before, with the call to `try body()` on a separate line.

Other than that little bug, I think this is reasonable enough, but I do think the API naming might be an issue. We'll see what @weissi thinks.

Reply:
Ah, sadly I totally missed your point 😂 It is very helpful to me, let me change it immediately. Yeah, I tried but failed to find a good name, I'd like to hear you guys opinions.